### PR TITLE
[ISSUE #3366] Allocate MessageQueue to client first by same Machine IP.

### PR DESCRIPTION
**Pull request by ni-ze**
add a new allocate MessageQueue strategy: Allocated by ip, if messageQueue localed in broker has same ip with client, Allocate messageQueue to this client firstly.

### CODE REVIEW

File: broker/src/main/java/org/apache/rocketmq/broker/processor/QueryAssignmentProcessor.java [hunk 1]
```diff
@@ -220,13 +220,13 @@ private RemotingCommand queryAssignment(ChannelHandlerContext ctx, RemotingComma
                         } else {
                             if (cidAll.size() <= mqAll.size()) {
                                 //consumer working in pop mode could share the MessageQueues assigned to the N (N = popWorkGroupSize) consumer following it in the cid list
-                                allocateResult = allocateMessageQueueStrategy.allocate(consumerGroup, clientId, mqAll, cidAll);
+                                allocateResult = allocateMessageQueueStrategy.allocate(null, consumerGroup, clientId, mqAll, cidAll);
```
Comment (ShannonDing) [inline broker/src/main/java/org/apache/rocketmq/broker/processor/QueryAssignmentProcessor.java:223]: Is it compatible with the previous version?

**Comment by ShannonDing** [comment]
Why not add an interface, but directly modify the original interface?

File: broker/src/main/java/org/apache/rocketmq/broker/processor/QueryAssignmentProcessor.java (hunk 1 above)
Comment (ni-ze) [reply broker/src/main/java/org/apache/rocketmq/broker/processor/QueryAssignmentProcessor.java:223]: QueryAssignmentProcessor and AllocateMessageQueueStrategy always  has same version, new version of QueryAssignmentProcessor will not depend on old AllocateMessageQueueStrategy interface.

**Comment by ni-ze** [comment]
> Why not add an interface, but directly modify the original interface?

when allocate MessageQueue, I think it is a natural thing to tall allocator which broker messageQueues local in.

**Comment by RongtongJin** [comment]
Hi @ni-ze  Could you resubmit your PR to the 5.0.0-beta branch because 5.0.0-alpha already release?

**Pull request closed without merging**